src/DeviceInput/deviceInputSystem.ts

```diff
@@ -125,7 +152,54 @@ export class DeviceInputSystem implements IDisposable {
         window.removeEventListener("gamepaddisconnected", this._gamepadDisconnectedEvent);
     }
 
+    /**
+     * Checks for existing connections to devices and register them, if necessary
+     * Currently handles gamepads and mouse
+     */

```

Comment:
Sorry I wasn't clear. I meant add comments to the contract so that people using the callbacks know that onConnected events for input manager / device source manager will not fire for keyboards.

Reply:
I'm not sure I understand what you mean.  With how the code is set up, it will still fire the connected callback/observables with the first input of a keyboard (and any changes to the connected obserables or DIS callback).  Did you, by any chance, mean the onDisconnected events?  The only difference between keyboards/mice and other input methods is that they don't ever trigger a disconnect scenario.

Reply:
We chatted offline and straightened things out.